### ISSUE TITLE
{{#each}} helper does not update on Array Mixin

### ISSUE THREAD

**Issue by ssured** [emberjs/ember.js#10650]
It seems the {{#each}} helper does not update on changes to objects implementing the Array or MutableArray mixin.

See http://emberjs.jsbin.com/gawiquhate/1/edit?html,js,output

**Comment by lolmaus** [comment]
That's a duplicate of #10405, though the solution suggested there does not work in your JSBin, so the issue is legit.

@MiguelMadero, any suggestions?

**Comment by stefanpenner** [comment]
@wycats although we want to kill this, we do likely need a way to broadcast this arbitrary enumerable/iterable has changed re-render me.

**Comment by rwjblue** [comment]
My suggestion in "post glimmer" here would be to use a helper that properly subscribes to the array change events and recomputes:

```hbs
{{#each (rerender-on-array-change model) as |item|}}
{{/each}}
```

A class based helper (aka `Ember.Helper.extend()`) that subscribed to its first argument could always recompute and return a newly built array (forcing an intelligent rerender).  This is roughly what we did for reduced computed type things, and it has worked pretty well.

I'm going to close this for now, in favor of the solution I outlined above.
